kotlin-playground/src/main/kotlin/org/gonza/kotlinplayground/domain/LottoNumber.kt

```diff
@@ -0,0 +1,13 @@
+package org.gonza.kotlinplayground.domain
+
+import org.gonza.kotlinplayground.enum.LottoRange
+
+data class LottoNumber(
+    val value: Int = (LottoRange.START.value..LottoRange.END.value).random(),
+) {
+    init {
+        require(value in LottoRange.START.value..LottoRange.END.value) { "로또 번호는 1부터 45까지의 숫자여야 합니다." }
+    }
```

Comment:
```suggestion
data class LottoNumber private constructor(val value: Int) {
    companion object {
        fun create(number: Int): LottoNumber {
            require(LottoRange.isValid(number)) { "로또 번호는 1부터 45까지의 숫자여야 합니다." }
            return LottoNumber(number)
        }
        
        fun random(): LottoNumber {
            return LottoNumber(LottoRange.randomNumber())
        }
    }
}
```

이런식으로 수정을 하면 어떨까요? 
지금은 전달된 value가 없으면 랜덤값을 만들고, value가 있다면 그 값이 우선적으로 사용되고 기본값은 무시가 될텐데, 자동 로또 발급을 염두해두고 하신걸까요?

또한 `LottoRange.START.value..LottoRange.END.value` 부분은 LottoRange에서 검증 메서드를 만드는 것이 책임이 더 어울려 보입니다!

Reply:
저는 객체 지향적으로 클래스를 쓰려면 최대한 생성자를 써야 한다는 주의라서 create, random 등 메소드가 추가되면 개발자나 사용하는 입장에서 어떤 메소드를 써야 하는지 다 들어가서 검증해야 한다고 생각합니다

Reply:
또한 LottoRange.START.value..LottoRange.END.value 부분은 LottoRange에서 검증 메서드를 만드는 것이 책임이 더 어울려 보입니다! -> 👍 반영완!